modules/stochastic_tools/include/state_sim/model/actions/StateSimAction.h

```diff
@@ -0,0 +1,38 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#ifndef STATESIMACTION_H
+#define STATESIMACTION_H
+
+#include "StateSimBase.h"
+#include <string>
+
+class StateSimAction : public StateSimBase
+{
+protected:
+  ACTION_TYPE_ENUM _act_type;
+
+private:
+  static unsigned int _counter;
```

Comment:
This is not for statistics it is for a unique IDs per model being run. For use in bitsets so they need to start at 1 and use every slot.

This will be fine in multi app because each model runs separately and unrelated instances with the same IDs are ok.  I don't see an issue with multi-threading do you?